### PR TITLE
Foundations: Remove reference to the show Silicon Valley (tabs vs spaces)

### LINKED ISSUE #28682: Clean Code: <Tabs vs Spaces link leads to a private video>
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [ ] Would you like to work on this issue?

### Describe your suggestion

Change the link to one that works.

What it currently looks like (at least for me):
image

### Path

Foundations

### Lesson Url

https://www.theodinproject.com/lessons/foundations-clean-code

### (Optional) Discord Name

_No response_

### (Optional) Additional Comments

_No response_

### PR DESCRIPTION
**Pull request by mathdebate09**
## Because

The video was unavailable and rather unhelpful and had a constant risk of being taken down, thus removing it.

## This PR

- Remove the video link
- Add a Reddit discussion

## Issue

Closes #28682

## Additional Information

I'm  somewhat doubtful of the reddit discussion, let me know if needs to be removed, lgtm

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project curriculum contributing guide](https://github.com/TheOdinProject/curriculum/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

File: foundations/javascript_basics/clean_code.md [hunk 1]
```diff
@@ -148,7 +148,7 @@ You might wonder why this variable is declared with all caps when we recommended
 
 ### Indentation and line length
 
-Now it's time to head to more controversial topics. The war between coders that use tabs and coders that use spaces to indent their code is essentially a joke by now, as demonstrated in the [tabs versus spaces scene from the show Silicon Valley](https://www.youtube.com/watch?v=SsoOG6ZeyUI).
+Now it's time to head to more controversial topics. The war between coders that use tabs and coders that use spaces to indent their code is essentially a [joke](https://www.reddit.com/r/programming/comments/p1j1c/tabs_vs_spaces_vs_both/) by now.
```
Comment (MaoShizhong) [inline foundations/javascript_basics/clean_code.md:151]: Rephrase suggestion to allow for a more accessible link text label

```suggestion
Now it's time to head to more controversial topics. There's already a joke about the [war between coders who indent with spaces versus tabs](https://www.reddit.com/r/programming/comments/p1j1c/tabs_vs_spaces_vs_both/).
```

Comment (ruben-vl) [reply foundations/javascript_basics/clean_code.md]: I find this wording "There's a joke about" to have a different meaning than "essentially being a joke".

I would change it to "The [war betwen ... with spaces versus tabs](link) is essentially a joke."

Comment (MaoShizhong) [reply foundations/javascript_basics/clean_code.md]: My reading of the original "essentially a joke" wasn't mocking the spaces/tabs war, but that the situation literally is a joke in the community, which the suggested change maintains the meaning of.

File: foundations/javascript_basics/clean_code.md [hunk 2]
```diff
@@ -148,7 +148,7 @@ You might wonder why this variable is declared with all caps when we recommended
 
 ### Indentation and line length
 
-Now it's time to head to more controversial topics. The war between coders that use tabs and coders that use spaces to indent their code is essentially a joke by now, as demonstrated in the [tabs versus spaces scene from the show Silicon Valley](https://www.youtube.com/watch?v=SsoOG6ZeyUI).
+Now it's time to head to more controversial topics. There's already a joke about the [war between coders who indent with spaces versus tabs](https://www.reddit.com/r/programming/comments/p1j1c/tabs_vs_spaces_vs_both/).
```
Comment (MaoShizhong) [inline foundations/javascript_basics/clean_code.md:151]: On a second read, what do you think about the below? Sorry to be a hassle xD Just feel it needs more focus on things being light-hearted, which helps reinforce the "conventions, not absolutes" tone of the lesson.

```suggestion
Now it's time to head to more "controversial" topics (there's a joke about the [war between coders who indent with spaces versus tabs](https://www.reddit.com/r/programming/comments/p1j1c/tabs_vs_spaces_vs_both/)).
```

Comment (mathdebate09) [reply foundations/javascript_basics/clean_code.md:151]: This feels better to me, slightly more accessible but definitely makes a difference!

**Review by MaoShizhong — APPROVED** [review]

**Pull request merged**